src/app/libp2p_helper/src/codanet.go

```diff
@@ -495,11 +496,15 @@ func (h *Helper) handlePxStreams(s network.Stream) {
 
 		for _, peer := range peers {
 			go func() {
-				err = h.Host.Connect(h.Ctx, peer)
-				if err != nil {
-					logger.Errorf("failed to connect to peer err=%s", err)
+				connInfo := h.ConnectionManager.GetInfo()
+				h.Host.Peerstore().AddAddrs(peer.ID, peer.Addrs, peerstore.ConnectedAddrTTL)
```

Comment:
I think you only want to add them to the peerstore explicitly in the `else` case of the if statement (connecting to a peer implicitly adds the peer to the peerstore, from my understanding). May not actually hurt anything, but depends on the implementation of `AddAddrs`.